lib/SecureStorage.ts

```diff
@@ -0,0 +1,72 @@
+import * as ExpoSecureStore from "expo-secure-store"
+
+export type SecureStorage = typeof ExpoSecureStore
+
+export const SECURE_STORAGE_KEY_PREFIX = "@SecureStorage:"
+
+export class AmplifySecureStorage<Store extends SecureStorage> {
+  private syncPromise?: Promise<void>
+  private store: Store
+  private keyList: string[]
+  private cache = new Map<string, string>()
+
+  constructor (store: Store) {
+    this.store = store
+    this.keyList = []
+  }
+
+  setItem (key: string, value: string) {
+    if (!this.keyList.find((k) => k === key)) {
+      this.keyList.push(key)
+    }
+    this.cache.set(SECURE_STORAGE_KEY_PREFIX + key, value)
+    this.store.setItemAsync("KeyList", JSON.stringify(this.keyList))
```

Comment:
Nit: I would extract a constant for `"Keylist"` since you use it in different places.